Updating libabigail to 2.0

vsoch opened the pull request:
Libabigail is updated to 2.0, and this fix is important because the previously working libabigail analyzer doesn't appear to work. I can't figure out what changed between it and it's dependencies, but without the fix (it's a bit of hack) here, a missing symbol is used that I was not able to compile with any set of dependencies. The missing symbol is documented on line 7 here:

https://chromium.googlesource.com/external/elfutils/+/515dd0acc77673c953380bcf5ccfb05b83c5a3ab/NEWS and my error again:

And the error without the fix is:

```bash
 error: 'DW_LANG_PL1' was not declared in this scope; did you mean 'DW_LANG_PLI'
```

So this line https://sourceware.org/git/?p=libabigail.git;a=blob;f=src/abg-dwarf-reader.cc;h=1d6ad24cbfcc2d94c07311bb04112f14f4f0e71c;hb=HEAD#l11056 needs to be DW_LANG_PLI.  needs to be updated. from PL1 to PLI. I'll make sure to ping the libabigail folks about it, and update the hack here as soon as it's fixed in a new release.

But this should be important to merge because as far as I've tested, the current recipe no longer works, period.

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
updating libabigail to 2.0

it looks like we need additional deps when compiling from master, added here.

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

alalazo commented on var/spack/repos/builtin/packages/libabigail/package.py:
```diff
@@ -10,19 +10,41 @@ class Libabigail(AutotoolsPackage):
 
+    depends_on('autoconf', type='build', when='@master')
+    depends_on('automake', type='build', when='@master')
+    depends_on('libtool', type='build', when='@master')
```
If you think it helps readability (can be in another PR) this can be written as:
```python
with when('@master'):
    depends_on('autoconf', type='build')
    depends_on('automake', type='build')
    depends_on('libtool', type='build')
```

vsoch replied:
Oh nice! I actually think I’ll need to follow up with another PR to tweak the master install, so I can use this syntax then if you like.

alalazo commented on var/spack/repos/builtin/packages/libabigail/package.py:
```diff
@@ -10,19 +10,41 @@ class Libabigail(AutotoolsPackage):
+                content = fd.read()
+            content = content.replace("case DW_LANG_PL1:", "case DW_LANG_PLI:")
+            with open("src/abg-dwarf-reader.cc", "w") as fd:
+                fd.write(content)
```
Only one minor request: can we extract this in a `patch()` method?

vsoch replied:
How do I do that - make a commit and save the diff as a patch alongside the package?

vsoch replied:
actually I just figured it out! Going to test first. I can easily make the other changes too.

alalazo reviewed (changes requested):
One minor request, but otherwise LGTM.

vsoch commented:
@alalazo updated PR!
- added the changes as a patch instead (tested and seems to still work ok)
- removed the extra dependencies given master. Firstly, an Autotools package seems to come with them. Secondly, I'm going to do more testing of installing from master in the coming weeks, and I can PR when I have something good working (I haven't tested yet tbh, but for one of our projects we need a development environment with libabigail so will test soon!)

alalazo commented on var/spack/repos/builtin/packages/libabigail/package.py:
```diff
@@ -26,3 +32,11 @@ class Libabigail(AutotoolsPackage):
     depends_on('doxygen', type="build", when="+docs")
     depends_on('py-sphinx', type='build', when="+docs")
+
+    # The symbol PL1 needs to be renamed to PLI
+    patch("0001-plt.patch")
```
@vsoch To reply to your question: you can either do what you did here (absolutely fine) or you can add a:
```
def patch(self):
    pass
```
method to the package.

alalazo replied:
Also, if the patch is accepted upstream we can remove it from Spack in a following PR and use a url in the directive
